x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/last_value.tsx

```diff
@@ -0,0 +1,214 @@
+    if (!hasDateFields) {
+      return i18n.translate('xpack.lens.indexPattern.lastValue.disabled', {
+        defaultMessage: 'This function requires the presence of a date field in your index',
+      });
+    }
+  },
+  buildColumn({ field, previousColumn, indexPattern }) {
+    const sortField =
+      indexPattern.timeFieldName ||
+      indexPattern.fields.filter((f) => f.type === 'date')[0].name ||
+      field.name;
+
+    const sortOrder =
+      (previousColumn?.params &&
+        'sortOrder' in previousColumn.params &&
+        previousColumn?.params?.sortOrder) ||
+      'desc';
+
+    return {
+      label: ofName(field.displayName),
+      dataType: field.type as DataType,
+      operationType: 'last_value',
+      isBucketed: false,
+      scale: 'ratio',
+      sourceField: field.name,
+      params: {
+        sortOrder,
+        sortField,
+      },
+    };
+  },
+  toEsAggsConfig: (column, columnId) => ({
+    id: columnId,
+    enabled: true,
+    schema: 'metric',
+    type: 'top_hits',
+    params: {
+      field: column.sourceField,
+      aggregate: 'concat',
+      size: 1,
+      sortOrder: column.params.sortOrder,
+      sortField: column.params.sortField,
+    },
+  }),
+
+  isTransferable: (column, newIndexPattern) => {
+    const newField = newIndexPattern.getFieldByName(column.sourceField);
+    return Boolean(newField && newField.type === column.dataType);
+  },
+
+  paramEditor: ({ state, setState, currentColumn, layerId }) => {
+    const dateFields = getDateFields(state.indexPatterns[state.layers[layerId].indexPatternId]);
+    const sortOrderButtons = [
+      {
+        id: `lns-lastValue-ascending`,
+        label: 'Ascending',
+        value: 'asc',
+      },
+      {
+        id: `lns-lastValue-descending`,
+        label: 'Descending',
+        value: 'desc',
+      },
+    ];
+    return (
+      <>
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.lastValue.sortField', {
+            defaultMessage: 'Sort by date field',
+          })}
+          display="columnCompressed"
+          fullWidth
+        >
+          <EuiSelect
+            compressed
+            data-test-subj="lns-indexPattern-lastValue-sortField"
+            options={dateFields?.map((field: IndexPatternField) => {
+              return {
+                value: field.name,
+                text: field.displayName,
+              };
+            })}
+            value={currentColumn.params.sortField}
+            onChange={(e: React.ChangeEvent<HTMLSelectElement>) =>
+              setState(
+                updateColumnParam({
+                  state,
+                  layerId,
+                  currentColumn,
+                  paramName: 'sortField',
+                  value: e.target.value,
+                })
+              )
+            }
+            aria-label={i18n.translate('xpack.lens.indexPattern.terms.orderBy', {
+              defaultMessage: 'Order by',
+            })}
```

Comment:
I'm confused by this... The visible label is "Sort by date field" but this one is different. Is this one needed? Is the visible label incorrect?

Reply:
Hey @myasonik, yes, too much copy-paste, correcting it!